internal/scheduler/scheduling_algo_test.go

```diff
@@ -70,24 +70,41 @@ func TestSchedule(t *testing.T) {
 			queuedJobs:               testfixtures.N16Cpu128GiJobs(testfixtures.TestQueue, testfixtures.PriorityClass3, 10),
 			expectedScheduledIndices: []int{0, 1},
 		},
+		"schedule onto executors with some unacknowledged jobs": {
+			schedulingConfig: testfixtures.WithMaxUnacknowledgedJobsPerExecutorConfig(16, testfixtures.TestSchedulingConfig()),
```

Comment:
If the executor has 16 unacknowledged jobs and `MaxUnacknowledgedJobsPerExecutor` is set to 16, then we give it more jobs.